### ISSUE TITLE
Motivation for the new signal streams?

### ISSUE THREAD

**Issue by jpgill86** [NeuralEnsemble/python-neo#992]
Hello!

After spending a lot of time reading closed issues, merged pull requests, and code, I am still uncertain I understand the motivation for the new concept of signal streams, which will be released with the upcoming version 0.10.0.

Prior to the introduction of signal streams, raw IOs had the capability to specify a `group_id` for each signal channel. With `signal_group_mode = "group-by-same-units"` (default since 0.9.0), signals with common characteristics (i.e., sampling rate, data type, and group id) were combined into one multi-channel AnalogSignal, and a Group was created for them (caveat: if the group contained signals with different units, these were split into subgroups).

Signal streams seem to be a replacement for the old concept of signal groups. One difference I have noticed is that, whereas the old signal groups could only have an id, the new signal streams can be named, which can certainly be useful. Otherwise, I don't understand how they differ. Were stream names the primary motivation? Or perhaps there was a performance gain obtained by refactoring?

If streams are essentially just nameable signal groups, why not continue to call them signal groups, in accordance with the Group class that they are analogous to? Why introduce this new term "stream"? The change seems to indicate there are important differences, and consequently I have spent a lot of time searching for them (in vain, haha 😅).

Are there important aspects that I have overlooked?

**Comment by samuelgarcia** [comment]
Salut Jeffrey,
yes indeed, this signal_tream is totally the replacement of this "group_id"  (same signal caracteristics).
But now there is explicitly a structure in header for that.

Main motivations for this:
  * handle annonations for this "streams" (aka group)
  * handle array_annotation. Needed for long time)
  * better API for other project that are "mono stream" (spikeinterface)

About the movitation for naming, I don't know it could be "group_id" also like before.
But now, this stream_index is part signature of: `get_signal_size()/get_signal_t_start()/get_analogsignal_chunk()`.
In was not the case before.
I wanted to be sure that developer would focus on this small change!!
A change in naming "group > stream" ensure that accustomed devs focus on it.
And it works :)

In short, concept have not change. API have change just a little. Name have change.
This rawio is rather internal, so the impact is low for users.
Except for spikeinterface devs and ephyviwer of course.

@PeterNSteinmetz was in the same feeling than you. If you want to a petition to come back on "group_id" I am open héhé.

**Comment by PeterNSteinmetz** [comment]
My view is that group_id is a somewhat better name and since the argument name is different than the names in the prior signature, that should work. If it is going to be changed, certainly should be done prior to the 0.10 release.

However, even more important than that name change, I think, is incorporating PR #970 which considerably clarifies this entire idea of grouping and makes it more explicit.

**Comment by jpgill86** [comment]
\*smacks forehead many times\*

I can't believe I overlooked @PeterNSteinmetz's issue (#969) and pull request (#970) on exactly this topic! I will check them out when I have more time. At a glance, it looks like this issue is redundant, so I will close it.
